.jenkins/pipelines/nightly.Jenkinsfile

```diff
@@ -61,18 +78,26 @@ pipeline {
                     }
                     axis {
                         name 'TEST_PIPELINE'
+                        // Append to the list below to add a new test
                         values 'Unit', 'Solutions', 'DotNet', 'DotNet-P1', 'Azure-SDK', 'PyTorch', 'OpenMP-Testsuite'
                     }
+                    axis {
+                        name 'VM_GENERATION'
+                        values 'v3', 'v2'
+                    }
                 }
                 stages {
                     stage("Matrix") {
                         steps {
                             script {
-                                // Workaround for skipping .NET P1 tests as dynamic matrix axis values are not supported
-                                // https://issues.jenkins.io/browse/JENKINS-62127
-                                if ( ! ( (TEST_PIPELINE == 'DotNet-P1' && ! params.RUN_DOTNETP1)
-                                      || (TEST_PIPELINE == 'OpenMP-Testsuite' && ! params.RUN_OPENMP_TESTSUITE) )) {
-                                    stage("${OS_VERSION} ${TEST_PIPELINE} (${TEST_CONFIG})") {
+                                stage("${TEST_PIPELINE} ${OS_VERSION} ACC-${VM_GENERATION} ${TEST_CONFIG}") {
+                                    // Workaround for skipping optional tests as dynamic matrix axis values are not supported
+                                    // https://issues.jenkins.io/browse/JENKINS-62127
+                                    if ( OPTIONS_TO_SKIP.contains(TEST_PIPELINE) || OPTIONS_TO_SKIP.contains(VM_GENERATION) ) {
```

Comment:
Just being nitpicky, but it might be a bit more readable if tests and VMs to ignore were kept separate, eg. `IGNORE_TESTS` and `IGNORE_VM_GENERATION`, instead of having them in the same list.

Reply:
Thanks for the suggestion. I've implemented this.